### ISSUE TITLE
Builder examples are not tested don't fully explain that order maters

### ISSUE THREAD

**Issue by TDHolmes** [jamwaffles/ssd1306#87]
Looking at the documentation for the Builder module, it states I can do the following:

```
use ssd1306::prelude::*;
use ssd1306::Builder;

fn main() {
    let mut disp: GraphicsMode<_> = Builder::new()
        .connect_i2c(i2c)
        .with_size(DisplaySize::Display128x32)
        .into();
}
```

However, I get the following error:
```
error[E0599]: no method named `with_size` found for type `ssd1306::mode::displaymode::DisplayMode<ssd1306::mode::raw::RawMode<ssd1306::interface::i2c::I2cInterface<hal::sercom::I2CMaster3>>>` in the current scope
  --> src/main.rs:49:10
   |
49 |         .with_size(DisplaySize::Display128x32)
   |          ^^^^^^^^^
```

The documentation should be fixed and enable testing on them so this doesn't happen again

**Comment by TDHolmes** [comment]
Oh I think I see the issue - looks like you have to connect the comms as the last step. However, I think the issue of them not being tested is still valid - adding some hidden boilerplate will enable these examples to be tested.

Also, documenting that the connect step has to come last would be good.

**Comment by jamwaffles** [comment]
Ouch, sorry that you found the example confusing! I'll take a look at why they're not tested and I'll add some docs in to clarify the ordering. Thanks for pointing these issues out!

**Comment by TDHolmes** [comment]
No worries, I'm liking the driver now that I have something working! This and #88 had me confused for a bit though haha.

**Comment by jamwaffles** [comment]
Fixed and released in [0.2.6](https://crates.io/crates/ssd1306/0.2.6). Thanks for raising this issue!
